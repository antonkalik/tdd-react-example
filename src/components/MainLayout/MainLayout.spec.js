import { render, screen } from "@testing-library/react";
import { MainLayout } from "./index";

jest.mock("src/components/Header", () => ({
  Header: () => <div data-testid="header">Header</div>,
}));

jest.mock("src/components/Footer", () => ({
  Footer: () => <div data-testid="footer">Footer</div>,
}));

describe("<MainLayout />", () => {
  test("should have header", () => {
    render(<MainLayout />);
    expect(screen.getByTestId("header")).toBeInTheDocument();
  });

  test("should have footer", () => {
    render(<MainLayout />);
    expect(screen.getByTestId("footer")).toBeInTheDocument();
  });

  test("should have children", () => {
    render(
      <MainLayout>
        <div data-testid="children">Children</div>
      </MainLayout>
    );
    expect(screen.getByTestId("body")).toBeInTheDocument();
    expect(screen.getByTestId("children")).toBeInTheDocument();
  });

  test("should render", () => {
    const { container } = render(<MainLayout />);
    expect(screen.getByTestId("main-layout")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
