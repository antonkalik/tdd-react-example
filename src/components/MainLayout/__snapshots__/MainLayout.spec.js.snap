// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainLayout /> should render 1`] = `
<div>
  <div
    data-testid="main-layout"
  >
    <div
      data-testid="header"
    >
      Header
    </div>
    <div
      data-testid="body"
    />
    <div
      data-testid="footer"
    >
      Footer
    </div>
  </div>
</div>
`;
